checkstyle/import-control.xml

```diff
@@ -48,6 +48,8 @@
   <allow pkg="org.apache.kafka.common.utils" />
   <allow pkg="org.apache.kafka.common.errors" exact-match="true" />
   <allow pkg="org.apache.kafka.common.memory" />
+  <!-- anyone can use persistent collection factories/non-library-specific wrappers -->
+  <allow pkg="org.apache.kafka.pcoll" exact-match="true" />
```

Comment:
Well, "anyone" on  the server side :) I don't think we want the kafka client to take this dependency. At least not yet, until we have a very clear use-case in mind.

So with that in mind, sadly it might be better to add this "allow" to server-common, metadata, and core individually, rather than up here. (although that's slightly more work I realize)